polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -892,8 +892,11 @@ void MemorySelector::analyze(LoopLikeOpInterface loop, AccessKind accessKind) {
 
     std::optional<MemoryAccess> memAccess =
         memAccessAnalysis.getMemoryAccess(memRefAccess);
-    assert(memAccess.has_value() &&
-           "Expecting valid memory access analysis result");
+    if (!memAccess.has_value()) {
```

Comment:
In `isCandidateAccess()`, we should have considered `memRefAccess` not a candidate, why is it still enter `analyze()`?
```
  std::optional<MemoryAccess> memAccess =
      memAccessAnalysis.getMemoryAccess(memRefAccess);
  if (!memAccess.has_value())
    return false;
```